src/configure/clients/github/githubClient.ts

```diff
@@ -1,69 +1,150 @@
+            })
+                .then((detail: GitHubRepo) => {
+                    return detail;
+                }).catch(error=>{
+                        return null;
+                });
+        }
+        catch(error){
+            return null;
+        }
     }
 
-    private async _getGitHubSecretKey() : Promise<GitHubSecretKey> {
-        let request = <UrlBasedRequestPrepareOptions> {
-                url: GitHubProvider.getFormattedGitHubApiUrlBase(this.url) + "/actions/secrets/public-key",
+    public async listOrganizations(PATtoken: string, forceRefresh?: boolean): Promise<void | GitHubOrganization[]> {
+        if (!this.listOrgPromise || forceRefresh) {
+            let restClient = new RestClient();
+            this.listOrgPromise = restClient.sendRequest(<UrlBasedRequestPrepareOptions>{
+                url: "https://api.github.com/user/orgs",
                 method: 'GET',
                 headers: {
+                    "Authorization": "Bearer " + PATtoken,
                     "User-Agent": UserAgent,
                     "Content-Type": "application/json",
-                    "Authorization": "Bearer " + this.patToken,
                     "Accept": "*/*"
                 },
-                serializationMapper: null,
-                deserializationMapper: null
-            }
-        let restClient = new RestClient();
-        return (await restClient.sendRequest(request)) as GitHubSecretKey;
+                deserializationMapper: null,
+                serializationMapper: null
+            })
+                .then((organizations: Array<GitHubOrganization>) => {
+                    organizations = organizations.sort((org1, org2) => stringCompareFunction(org1.login, org2.login));
+                    return organizations;
+                });
+        }
+        return this.listOrgPromise; 
     }
 
-    private async _setGithubSecret(secretName: string, key_id: string, encrypted_secret: string): Promise<void> {
+    public async generateGitHubRepository(orgName: string, localPath: string): Promise<GitHubRepo | void>{
+        let repoName = localPath.substring(localPath.lastIndexOf("\\")+1).replace("*","-"); //need to include all the special characters here
+        let repoDetails = await this.createGithubRepo(orgName, repoName) as unknown as GitHubRepo | void;
+        // Case : GitHub Repo name is same as the local repo
+        if(repoDetails){
+            return repoDetails; 
+        }
+        //Case: Local repo name is not available, therefore organization name has been appended
+        repoName = repoName+"_"+orgName;
+        repoDetails = await this.createGithubRepo(orgName, repoName) as unknown as GitHubRepo | void;
+        if(repoDetails){
+            return repoDetails; 
+        }
+        //Case: If the above two names are not available, uuid is also appended
+        return await this.createGithubRepo(orgName, repoName+"-"+uuid().substr(0,5));   //need to add proper Uiid or goid
```

Comment:
you can put spaces to increase readability. It is not bash :smile:

Reply:
Sure, will keep in mind from now on to make the code more readable :)